lib/decimal.ex

```diff
@@ -1388,6 +1388,45 @@ defmodule Decimal do
     tmp
   end
 
+  @doc """
+  Returns `true` when the given `decimal` has no significant digits after dot.
+
+  ## Examples
+
+      iex> Decimal.new("1.00") |> Decimal.integer?()
+      true
+
+      iex> Decimal.new("1.1000") |> Decimal.integer?()
+      false
+  """
+  @spec integer?(t) :: boolean
+  def integer?(%Decimal{} = decimal),
+    do: not fraction?(decimal)
+
+  @doc """
+  Returns `true` when the given `decimal` has significant digits after dot.
+
+  ## Examples
+
+      iex> Decimal.new("1.1000") |> Decimal.fraction?()
+      true
+
+      iex> Decimal.new("1.00") |> Decimal.fraction?()
+      false
+  """
+  @spec fraction?(t) :: boolean
+  def fraction?(%Decimal{exp: exp} = decimal),
+    do: exp < 0 and number_after_dot(decimal) != 0
+
+  defp number_after_dot(%Decimal{coef: coef, exp: exp}) do
+    ten_power_exp =
+      10
+      |> :math.pow(Kernel.abs(exp))
```

Comment:
This is not safe since `:math.pow/1` works on floating numbers, you can use the private function `pow10/1` instead.

Reply:
@ericmj Thanks! Done! ✅